src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -335,26 +341,23 @@ public void add(@CheckForNull Content content) {
         }
         toProcess.add(new StringContent("manifest.md", manifest.toString()));
         try {
-            ZipOutputStream zip = new ZipOutputStream(outputStream);
-            try {
-                BufferedOutputStream bos = new BufferedOutputStream(zip, 16384) {
-                    @Override
-                    public void close() throws IOException {
-                        // don't let any of the contents accidentally close the zip stream
-                        super.flush();
-                    }
-                };
+            try (ZipArchiveOutputStream zip = new ZipArchiveOutputStream(new BufferedOutputStream(outputStream, 16384))) {
+                final FilteredOutputStream out = new FilteredOutputStream(new IgnoreCloseOutputStream(zip), filter);
                 while (!toProcess.isEmpty()) {
-                    Content c = toProcess.poll();
+                    final Content c = toProcess.poll();
                     if (c == null) {
                         continue;
                     }
-                    final String name = c.getName();
+                    final String name = filter.filter(c.getName());
+                    final ZipArchiveEntry entry = new ZipArchiveEntry(name);
+                    entry.setTime(c.getTime());
                     try {
-                        ZipEntry entry = new ZipEntry(name);
-                        entry.setTime(c.getTime());
-                        zip.putNextEntry(entry);
-                        c.writeTo(bos);
+                        zip.putArchiveEntry(entry);
+                        if (name.endsWith(".png")) {
```

Comment:
The png files are graphs. There are associated csv files. Also, text files include .log, .log.1, .log.2, etc., along with some other extensions. Seemed easier to make a blacklist.

Reply:
I wonder if a decent way to avoid issues would be to check the first X bytes (maybe 10) and if there are ascii control characters other than tab, line feed, vertical tab, form feed, and carriage return (`0x09`-`0x0D`) don't filter the file. On a somewhat related note I am worried about mangling files using encodings other than UTF-8.

Reply:
Yes exactly, all of this will go wrong pretty quickly if things aren't UTF-8. Maybe a good idea to look at the components that use FileContent to see if any of their encodings can be predicted based on other information (e.g. will agent logs always use `Charset#defaultEncoding` from the agent? If so we could use that encoding for that file)

Reply:
Perhaps it may be worth refactoring a bit to use Writers instead of OutputStreams where possible?

Reply:
This is making me think that some cleanups in SupportPlugin would be worth it just for the better charset usage.